wallet/src/actors/app/methods.rs

```diff
@@ -314,10 +314,14 @@ impl App {
 
     /// Load a wallet's private information and keys in memory.
     pub fn unlock_wallet(
-        &self,
+        &mut self,
         wallet_id: String,
         password: types::Password,
     ) -> ResponseActFuture<types::UnlockedWallet> {
+        // If there is a synchronization from a previous session running, set unsyncing to false so that it can stop
+        if let Some(wallet) = self.state.get_current_wallet_session(wallet_id.clone()) {
+            wallet.unset_syncing().expect("Lock error")
```

Comment:
Do we have any guarantee that the other session will read this flag before the new session starts synchronizing? Otherwise, as soon as the new session starts synchronizing it will set the flag back to true, and we are have the same problem again.

Reply:
Is there one `syncing` flag for each session? In that case there is no problem.

Reply:
AFAIK yes